src/Config/Parser.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WPML\PB\Config;
+
+use WPML\FP\Lst;
+use WPML\FP\Obj;
+
+class Parser {
+
+	/** @var string $configRoot */
+	private $configRoot;
+
+	/** @var string $defaultConditionKey */
+	private $defaultConditionKey;
+
+	public function __construct( $configRoot, $defaultConditionKey ) {
+		$this->configRoot          = $configRoot;
+		$this->defaultConditionKey = $defaultConditionKey;
+	}
+
+	/**
+	 * Receives a raw config array (from XML) and convert it into
+	 * a page builder configuration array.
+	 *
+	 * @see WPML_Elementor_Translatable_Nodes::get_nodes_to_translate()
+	 *
+	 * @param array $allConfig
+	 *
+	 * @return array
+	 */
+	public function extract( array $allConfig ) {
+		$pbConfig = [];
+		$allWidgets = Obj::path( [ 'wpml-config', $this->configRoot, 'widget' ], $allConfig );
```

Comment:
@strategio I think you can use `pathOr( [], .... )` here to return an empty array and then remove the `if( $allWidgets )` check

Reply:
Done!